### PR TITLE
feat: SSR study mode related verses

### PR DESCRIPTION
**Pull request by afifvdin**
## Summary

Adds SSR (Server-Side Rendering) support for the Study Mode Related Verses tab, ensuring related verses data is pre-fetched and rendered on the server for better SEO and initial page load performance.

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] ♻️ Refactoring (no functional changes)

## Scope Confirmation

- [x] This PR addresses **one** feature/fix only
- [x] If multiple changes were needed, they are split into separate PRs

## Rollback Safety

- [x] Can be safely reverted without data issues or migrations

## Test Plan

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Manual testing performed

**Testing steps:**

1. Navigate to any verse's related verses page (e.g., `/1:1/related-verses`)
2. Verify that the related verses tab content is rendered in the initial HTML (can be confirmed via `curl` or View Page Source)
3. Verify infinite scroll pagination works correctly by scrolling down to load more related verses
4. Navigate to a different verse and confirm the related verses list resets properly

### Edge Cases Verified

- [x] ⏳ Loading state handled
- [x] ❌ Error state handled
- [x] 📭 Empty state handled
- [ ] 👤 Logged-in vs guest behavior (if applicable)

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
- [x] No `any` types used (or justified if unavoidable)
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [ ] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [x] Build succeeds (`yarn build`)
- [x] Edge cases and error scenarios are handled

### Documentation

- [x] Code is self-documenting with clear naming
- [ ] README updated (if adding features or setup changes)
- [x] Inline comments added for complex logic

### Localization (if UI changes)

- [x] All user-facing text uses `next-translate`
- [x] Only English locale files modified (Lokalise handles others)
- [x] RTL layout verified

### Accessibility (if UI changes)

- [x] Semantic HTML elements used
- [x] ARIA attributes added where needed
- [x] Keyboard navigation works

## AI Assistance Disclosure

- [ ] AI tools were NOT used for this PR
- [x] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

### CODE REVIEW

File: src/pages/[chapterId]/related-verses.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  chapter,
+  verseNumber,
+  chapterId,
+  fallback,
+  verse,
+  versesResponse,
+}) => {
+  const { t, lang } = useTranslation('quran-reader');
+
+  const navigationUrl = getVerseReflectionNavigationUrl(`${chapterId}:${verseNumber}`);
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/pages/[chapterId]/related-verses.tsx]: fixed
